lib/utils/validateTypes.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+/**
+ * Checks if the value is a boolean or a Boolean object.
+ * @param {any} value
+ * @returns {boolean}
+ */
+function isBoolean(value) {
+	return typeof value === 'boolean' || value instanceof Boolean;
+}
+
+/**
+ * Checks if the value is a number or a Number object.
+ * @param {any} value
+ * @returns {boolean}
+ */
+function isNumber(value) {
+	return typeof value === 'number' || value instanceof Number;
+}
+
+/**
+ * Checks if the value is the language type of Object.
+ * @param {any} value
+ * @returns {boolean}
+ */
+function isObject(value) {
+	return value instanceof Object;
+}
```

Comment:
[suggestion] I wonder if this function should be `isPlainObject` (#5336) because `value instanceof Object` can be applied to almost all objects:

```console
$ node
Welcome to Node.js v16.2.0.
Type ".help" for more information.
> new Function() instanceof Object
true
> new RegExp() instanceof Object
true
> new String() instanceof Object
true
```

So, I think `isPlainObject` should be below according to #5336:

```js
function isPlainObject(value) {
	return Object.prototype.toString.call(value) === '[object Object]';
}
```

What do you think?

Reply:
@ybiquitous Yes, that should be `isPlainObject`. My change copies the original behavior but you're right that it allowed too many objects. I've made the change in 34f948fc.